rpcs3/rpcs3qt/debugger_frame.cpp

```diff
@@ -857,20 +858,43 @@ void debugger_frame::UpdateUnitList()
 		}
 	};
 
+	do
 	{
 		const QSignalBlocker blocker(m_choice_units);
 
+		std::shared_lock idm_lock(id_manager::g_mutex, std::defer_lock);
+
+		for (u32 i = 0; i < 20; i++)
+		{
+			if (idm_lock.try_lock())
+			{
+				break;
+			}
+
+			busy_wait();
+		}
+
+		if (!idm_lock)
+		{
+			// Enqueue the list update for when IDM becomes available
+			// Avoids a potential deadlock when a thread crashed while holding IDM mutex (it may crash the emulator later in Emu.Kill() etc, but GUI will work until then)
+			m_threads_created = -1;
+			break;
+		}
+
```

Comment:
there's no reason to block signals before this section I think.
Also, all the GUI changes as well as the sceNpTrophy changes should better be seperate PRs.
This is way too crowded with random changes

Reply:
It's not because of QT, it's so idm mutex won't get locked for a while.

Reply:
This is all on the main thread, so I don't know where this would help

Reply:
Emulation threads use idm mutex all the time,

Reply:
The SignalBlocker is only meant to disable the signal/slot logic while the dropdown is cleared and refilled

Reply:
Again, not about QT at all. If this was a std::this_thread::sleep_for call for example I had move it out of the mutex scope as well for the same reason.

Reply:
what are you even talking about

Reply:
It's bad for performance to lock a mutex for a while.